localstack/services/iam/provider.py

```diff
@@ -298,6 +308,80 @@ def get_service_linked_role_deletion_status(
         # TODO: test
         return GetServiceLinkedRoleDeletionStatusResponse(Status=DeletionTaskStatusType.SUCCEEDED)
 
+    def put_user_permissions_boundary(
+        self, context: RequestContext, user_name: userNameType, permissions_boundary: arnType
+    ) -> None:
+        if user := moto_iam_backend.users.get(user_name):
+            user.permissions_boundary = permissions_boundary
+        else:
+            raise NoSuchEntityException()
+
+    def delete_user_permissions_boundary(
+        self, context: RequestContext, user_name: userNameType
+    ) -> None:
+        if user := moto_iam_backend.users.get(user_name):
+            if hasattr(user, "permissions_boundary"):
+                delattr(user, "permissions_boundary")
+        else:
+            raise NoSuchEntityException()
+
+    def create_user(
+        self,
+        context: RequestContext,
+        user_name: userNameType,
+        path: pathType = None,
+        permissions_boundary: arnType = None,
+        tags: tagListType = None,
+    ) -> CreateUserResponse:
+        response = call_moto(context=context)
+        user = moto_iam_backend.get_user(user_name)
+        if permissions_boundary:
+            user.permissions_boundary = permissions_boundary
+            response["User"]["PermissionsBoundary"] = AttachedPermissionsBoundary(
+                PermissionsBoundaryArn=permissions_boundary,
+                PermissionsBoundaryType="Policy",
+            )
+        return response
+
+    def get_user(
+        self, context: RequestContext, user_name: existingUserNameType = None
+    ) -> GetUserResponse:
+        response = call_moto(context=context)
+        moto_user_name = response["User"]["UserName"]
+        moto_user = moto_iam_backend.users.get(moto_user_name)
+        # if the user does not exist or is no user
+        if not moto_user and not user_name:
+            access_key_id = extract_access_key_id_from_auth_header(context.request.headers)
+            sts_client = aws_stack.connect_to_service(
+                "sts",
+                aws_access_key_id=access_key_id,
+                aws_secret_access_key=INTERNAL_AWS_SECRET_ACCESS_KEY,
+            )
+            caller_identity = sts_client.get_caller_identity()
+            caller_arn = caller_identity["Arn"]
+            if caller_arn.endswith(":root"):
+                return GetUserResponse(
+                    User=User(
+                        UserId=context.account_id,
+                        Arn=caller_arn,
+                        CreateDate=datetime.now(),
+                        PasswordLastUsed=datetime.now(),
+                    )
+                )
+            else:
+                raise CommonServiceException(
+                    "ValidationError",
```

Comment:
This is a sender fault right?

Reply:
Yes, you are not supposed to call get_user without a user name if you (the sender) is authenticated as a user principal